### DISCUSSION TITLE
question

### DISCUSSION THREAD

**Discussion by nadongjin** [leondgarse/Keras_insightface#81] [Q&A]
Hello ^^
Long time no see.
I have a question about the normalize of the image in your code.
----------------------------------------------------------------------------------------------------------------
normalize=lambda xx: (tf.cast(tf.image.decode_image(xx, channels=3), "float32") - 127.5) * 0.0078125
----------------------------------------------------------------------------------------------------------------
As above, you did 0.0078125 = 1/128.
But when I looked at the insightface's code, it was 1/128 for face detection and 1/127.5 for face recognition.
Normally, normalize -1<=x<=1. I wonder if there is any other reason for 1/128 instead of 1/127.5.
Thank you.

Ms1mv3 image cleaning is too painful.^^
I don’t' know the reason.
Why is there so much data that separated one class from the dataset?

**Comment by leondgarse** [comment]
`0.0078125` is actually from the very early implementation of the [mxnet insightface](https://github.com/deepinsight/insightface/tree/master/recognition/arcface_mxnet/image_iter.py#L224). IMO, the model training is not that sensitive for this `0.5` difference, either `127.5, 1 / 127.5` or `128, 1 / 128` or `127.5, 1 / 128`, just keeping them same for both training and evaluation is OK.
Ya, manually cleaning the dataset is very brave! Haven't got time for my model cleaning...

**Reply by nadongjin** [reply]
I will send you a clean dataset at the end of this month.
I think this will only help me if you find a sota architecutre from Enjinier's point of view. ^^

**Reply by leondgarse** [reply]
Ya, it's still better using the raw dataset for comparing with others.  Besides, I have finished some running use `MagfaceLoss`, and achieves `IJBB 0.957449, IJBC 0.971212` using `EfficienNetV2S + AdamW`, and the training much stable than ArcFace at the beginning. May upload it soon, need some more tests.

**Reply by nadongjin** [reply]
I'm looking forward to it ^^

**Reply by leondgarse** [reply]
It's uploaded, with a higher result `IJBB 0.958325, IJBC 0.971212`. Enjoy.

**Comment by nadongjin** [comment]
You're a genius. :)
I am still working on data clean work.
It is questionable whether your current architecture and train techniques
can be further improved.
I think data improvement is the only way to improve your performance from
now on.
I will do my best to work hard and give it to you quickly.

I wonder if the magface algorithm is definitely stable during learning.
Once again, thank you for your efforts.

2022년 2월 24일 (목) 오후 4:37, leondgarse ***@***.***>님이 작성:

> It's uploaded, with a higher result IJBB 0.958325, IJBC 0.971212. Enjoy.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/leondgarse/Keras_insightface/discussions/81#discussioncomment-2241988>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADRFGOCPVZBXFS3T2ZK3SXDU4XNZ5ANCNFSM5OPD5ITA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
유빈아빠

**Reply by leondgarse** [reply]
It's just some repeating works of parameters tuning. Lol
Ya, I don't know if its' reproducible. Just finally a recipe can applying `AdamW` for `EfficientNetV2M`, the previous `AdamW + ArcFace` actually most time not working well for `V2M` in my tests...
